File: Source/IntegrationTests/TestFiles/LitTests/LitTest/git-issues/Inputs/BoxPkg.go
// An extern Go class that follows the contract in the reference manual's
// "Equality of extern types": it implements EqualsGeneric as identity, which is
// the equality Dafny assumes for a class. Dafny cannot generate this, because Go
// does not allow methods to be declared on another package's type.
package BoxPkg

type Box struct {
	dummy byte
}

type CompanionStruct_Box_ struct{}

var Companion_Box_ = CompanionStruct_Box_{}

func New_Box_() *Box {
	return &Box{}
}

func (_this *Box) Equals(other *Box) bool {
	return _this == other
}

func (_this *Box) EqualsGeneric(x interface{}) bool {
	other, ok := x.(*Box)
	return ok && _this.Equals(other)
}

File: Source/IntegrationTests/TestFiles/LitTests/LitTest/git-issues/Inputs/BoxPkg.js
// An ordinary extern JavaScript class: no equals method, so generic
// comparison used to crash on it. Dafny models the class with identity
// equality, which the runtime must fall back to.
let BoxPkg = (function() {
  let $module = {};

  $module.Box = class Box {
    constructor () {
    }
  };

  return $module;
})();
